rfcs/20210908-structural-sparsity-flag.md

```diff
@@ -0,0 +1,104 @@
+# Title of RFC
+
+| Status        | (Proposed)       |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [398](https://github.com/tensorflow/community/pull/NNN) (update when you have community PR #)|
+| **Author(s)** | Elena Zhelezina (elena.zhelezina@arm.com) |
+| **Sponsor**   | David Rim (davidrim@gmail.com)                 |
+| **Updated**   | 2021-09-13                                           |
+| **Obsoletes** | TF-RFC it replaces, else remove this header          |
+
+## Objective
+
+Indicate that model saved in tflite file has layers that have been pruned with structural sparsity.
+
+## Motivation
+
+The model can be pruned using structural pruning with TensorFlow Model Optimization Toolkit. This functionality has been added by this PR [Structural pruning](https://github.com/tensorflow/model-optimization/pull/743). The model pruned this way is usually saved into .tflite file. When the model is loaded from .tflite file, all tensors need to be checked to identify whether they are structurally pruned. Currently to get benefits of this, every tensor needs to be checked every time once it is loaded.
+
+We propose to add information whether tensors are structurally pruned to tflite file. This allows to speed-up inference time by moving this check to the conversion time once tflite is created.
+
+## User Benefit
+
+The user can identify which layers of the model have been structurally pruned without doing computationally intensive check every time when the model is loaded from .tflite file.
+
+## Design Proposal
+
+We propose to add an optional structure `StructuralSparsityParameters` to the structure [Tensor] (https://github.com/tensorflow/tensorflow/blame/master/tensorflow/lite/schema/schema.fbs#L195) that provides the parameters of structural pruning. If it is present, then the tensor is structurally pruned.
+
+```
+table StructuralSparsityParameters {
+  // For a tensor that is structurally pruned we save the type of the pruning.
+  // (m, n) means that in a block of n elements, m elements are zeros.
+  // If this field is NULL, then the tensor is not sparse.
+  sparsity_type_m_by_n:[int];
```

Comment:
If I'm reading this correctly this is true/false? MxN are not stored in the model, but set at initialization?

Is the current flag for sparsity not enough?